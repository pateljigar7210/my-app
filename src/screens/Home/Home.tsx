
import * as React from 'react';
import { View, Text, Linking, StyleSheet, SafeAreaView ,TouchableOpacity} from 'react-native';
import { NavigationProp } from '@react-navigation/native';
import { NearEarthObjectsEntity } from '../Login/types/AstronodListType';

interface HomeScreenProps {
    navigation: NavigationProp<any, any>,
    route: any
}

const Home = ({ route, navigation }: HomeScreenProps) => {
    const astronodInfo = route?.params?.data as NearEarthObjectsEntity;

    return (
        <SafeAreaView>

            <View style={styles.containner}>
                <Text>{astronodInfo ? 'name: ' + astronodInfo.name : ''}</Text>
                <TouchableOpacity onPress={async () => {
                    if (astronodInfo) {
                        await Linking.openURL(astronodInfo.nasa_jpl_url);
                    }
                }}>
                    <Text style={styles.linkStyle}>{astronodInfo ? 'nasa jpl uri: ' + astronodInfo.nasa_jpl_url : ''}</Text>

                </TouchableOpacity>
                <Text>{astronodInfo ? "is_potentially_hazardous_asteroid: "+ astronodInfo.is_potentially_hazardous_asteroid + '' : ''}</Text>

            </View>


        </SafeAreaView>
    );
}

export default Home;

const styles = StyleSheet.create({
    containner: {
      
      backgroundColor:'#FFF',
        padding: 30,
        alignSelf:'center',
        margin:30,
        borderRadius:8,
        elevation: 4,
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 1 },
        shadowOpacity: 0.8,
        shadowRadius: 1,
    },
    linkStyle: {
        color: 'red'
    }
});
