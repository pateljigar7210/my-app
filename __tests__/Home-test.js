import React from 'react';
import renderer from 'react-test-renderer';

import 'react-native-gesture-handler';
import  Home from '../src/screens/Home/Home';

global.fetch = jest.fn(() => new Promise(resolve => resolve()));
jest.mock('react-native-gesture-handler', () => {});

it('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});