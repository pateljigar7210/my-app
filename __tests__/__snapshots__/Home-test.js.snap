// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 8,
        "elevation": 4,
        "margin": 30,
        "padding": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 1,
      }
    }
  >
    <Text>
      
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "red",
          }
        }
      >
        
      </Text>
    </View>
    <Text>
      
    </Text>
  </View>
</RCTSafeAreaView>
`;
